src/Kestrel.Core/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -29,6 +29,8 @@ public static partial class HttpUtilities
         private const ulong _http10VersionLong = 3471766442030158920; // GetAsciiStringAsLong("HTTP/1.0"); const results in better codegen
         private const ulong _http11VersionLong = 3543824036068086856; // GetAsciiStringAsLong("HTTP/1.1"); const results in better codegen
 
+        private static readonly Encoding HeaderValueEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: true, throwOnInvalidBytes: true);
```

Comment:
I'm using the default value of `encoderShouldEmitUTF8Identifier` set on `Encoding.UTF8`: https://docs.microsoft.com/en-us/dotnet/api/system.text.utf8encoding?redirectedfrom=MSDN&view=netframework-4.7.2#remarks. I'm interested in `throwOnInvalidBytes` and it seems like the value of `encoderShouldEmitUTF8Identifier` shouldn't have any consequence since we use it to decode only.

Reply:
Is there a reason not to make encoderShouldEmitUTF8Identifier false other than it's not the default? If we ever start using this for response headers, we wouldn't want to write a BOM.

Reply:
I just picked the default arbitrarily. I wasn't planning on using this for response headers but I suppose we could set it to false for defensive purposes.

Reply:
If you want to go max speed
```csharp
[UTF8Encoding.cs#L53-L59](https://github.com/dotnet/coreclr/blob/13c4ab0742ee7f80fb1cb98bef95e106b0f9c83a/src/System.Private.CoreLib/shared/System/Text/UTF8Encoding.cs#L53-L63)

// Allow for de-virtualization (see https://github.com/dotnet/coreclr/pull/9230)
private sealed class UTF8EncodingSealed : UTF8Encoding
{
    public UTF8EncodingSealed() : base(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true) { }

    public override ReadOnlySpan<byte> Preamble => Array.Empty<byte>();
}

private static readonly UTF8EncodingSealed HeaderValueEncoding = new UTF8EncodingSealed();
```